server/src/main/java/org/elasticsearch/repositories/FilterRepository.java

```diff
@@ -78,10 +78,11 @@ public void initializeSnapshot(SnapshotId snapshotId, List<IndexId> indices, Met
     }
 
     @Override
-    public SnapshotInfo finalizeSnapshot(SnapshotId snapshotId, List<IndexId> indices, long startTime, String failure, int totalShards,
-                                         List<SnapshotShardFailure> shardFailures, long repositoryStateId, boolean includeGlobalState) {
-        return in.finalizeSnapshot(snapshotId, indices, startTime, failure, totalShards, shardFailures, repositoryStateId,
-            includeGlobalState);
+    public SnapshotInfo finalizeSnapshot(SnapshotId snapshotId, String repository, List<IndexId> indices, long startTime, String failure,
```

Comment:
We shouldn't have to pass the repository name to `finalize`, the repository must know its own name.